src/TLuaInterpreter.cpp

```diff
@@ -9897,41 +9973,67 @@ int TLuaInterpreter::getAllMapUserData( lua_State * L )
 
 int TLuaInterpreter::downloadFile( lua_State * L )
 {
-    string path, url;
-    if( ! lua_isstring( L, 1 ) )
-    {
-        lua_pushstring( L, "downloadFile: wrong argument type" );
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if( ! pHost ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "downloadFile: NULL Host pointer - something is wrong!" )
+                        .toUtf8().constData() );
+        return 2;
+    }
+
+    QString localFile;
+    if( ! lua_isstring( L, 1 ) ) {
+        lua_pushstring( L, tr( "downloadFile: bad argument #1 type (local filename as string expected, got %1!)" )
+                        .arg( luaL_typename( L, 1 ) )
+                        .toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
-        path = lua_tostring( L, 1 );
+    else {
+        localFile = QString::fromUtf8( lua_tostring( L, 1 ) );
     }
-    if( ! lua_isstring( L, 2 ) )
-    {
-        lua_pushstring( L, "downloadFile: wrong argument type" );
+
+    QString urlString;
+    if( ! lua_isstring( L, 2 ) ) {
+        lua_pushstring( L, tr( "downloadFile: bad argument #1 type (remote url as string expected, got %1!)" )
+                        .arg( luaL_typename( L, 2 ) )
+                        .toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
-        url = lua_tostring( L, 2 );
+    else {
+        urlString = QString::fromUtf8( lua_tostring( L, 2 ) );
     }
-    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
-    QString _url = url.c_str();
-    QString _path = path.c_str();
-    QNetworkRequest request = QNetworkRequest( QUrl( _url ) );
+
+    QUrl url = QUrl::fromUserInput( urlString );
+
+    if( ! url.isValid() ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "downloadFile: bad argument #2 value (url is not deemed valid), validation\n"
+                               "produced the following error message:\n%1." )
+                        .arg( url.errorString() )
+                        .toUtf8().constData() );
+        return 2;
+    }
+
+    QNetworkRequest request = QNetworkRequest( url );
+    // This should fix: https://bugs.launchpad.net/mudlet/+bug/1366781
+    request.setRawHeader( QByteArray( "User-Agent" ),
+                          QByteArray( QStringLiteral( "Mozilla/5.0 (Mudlet/%1%2)" )
+                                      .arg( APP_VERSION )
+                                      .arg( APP_BUILD )
+                                      .toUtf8().constData() ) );
 #ifndef QT_NO_OPENSSL
-    if ( _path.contains("https") )
-    {
+    if( url.scheme() == QStringLiteral( "https" ) ) {
```

Comment:
This is the core of the issue - `_path` was the local pathFileName not the remote url needed.